### PR TITLE
Fix regression preprocessing

### LINKED ISSUE #942: Bug in FedotRegressionPreprocessingStrategy
Bug in `update_data()` of `sklearn_filters.py` when trying get data from `old_features[mask]`. It happens after pipeline composing, when fedot are trying to fit pipeline at whole datam only when this pipeline contains any next operations `['ransac_lin_reg', 'ransac_non_lin_reg', 'isolation_forest_reg']`.

Output is `Boolean index did not match indexed array along dimension 0; ...`

Script for repeating the bug:
```
data_path = f'{fedot_project_root()}/cases/data/cholesterol/cholesterol.csv'

data = InputData.from_csv(data_path, task=Task(TaskTypesEnum.regression))

train, test = train_test_data_setup(data)
problem = 'regression'

builder = PipelineBuilder()
linear_pipeline = builder.add_node('linear').to_pipeline()

auto_model = Fedot(
      problem=problem,
      timeout=10,
      metric='mae',
      preset='fast_train',
      initial_assumption=linear_pipeline,
      logging_level=logging.DEBUG,
)

auto_model.fit(features=train)
```

### PR DESCRIPTION
**Pull request by YamLyubov**
The problem was that tuner was returning already fitted pipeline. And when final pipeline (after composition and tuning) was fitted on whole data, operations in nodes was not refitted. This was causing the error.

Now tuner returns unfitted pipeline.

Fixes #942

### CODE REVIEW

**Review by nicl-nno — APPROVED** [review]
В какой-то из тестов бы ещё добавить проверку что пайплане не-fitted после тюнинга.
